crates/ra_hir/src/expr.rs

```diff
@@ -792,7 +802,21 @@ impl ExprCollector {
             ast::ExprKind::Label(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::IndexExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
             ast::ExprKind::RangeExpr(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
-            ast::ExprKind::MacroCall(_e) => self.alloc_expr(Expr::Missing, syntax_ptr),
+            ast::ExprKind::MacroCall(e) => {
+                let macro_name = e.name().map(|nr| nr.as_name()).unwrap_or_else(Name::missing);
+                let path = e.path().and_then(Path::from_ast);
+                let expanded = if let Some(tt) = e.token_tree() {
```

Comment:
This is the place where we should ask db to expand the macro. Currrently, only expansion to the list of items is supported. We should add expansion to an expr.